src/vs/editor/contrib/hover/browser/hoverOperation.ts

```diff
@@ -196,7 +197,32 @@ export class HoverOperation<T> extends Disposable {
 			this._asyncIterable = null;
 		}
 		this._result = [];
-		this._setState(HoverOperationState.Idle, false);
+		this._options = undefined;
+		this._state = HoverOperationState.Idle;
+	}
+
+	public get options(): U | undefined {
+		return this._options;
 	}
+}
+
+class HoverRunOnceScheduler<U> extends Disposable {
+
+	private readonly _scheduler: RunOnceScheduler;
+
+	private _options: U | undefined;
 
+	constructor(runner: (args: U) => void, delay: number) {
+		super();
+		this._scheduler = this._register(new RunOnceScheduler(() => runner(this._options!), delay));
+	}
+
+	schedule(options: U, delay: number): void {
+		this._options = options;
+		this._scheduler.schedule(delay);
+	}
+
+	cancel(): void {
+		this._scheduler.cancel();
+	}
 }
```

Comment:
This seems like a generic class, which doesn't have anything to do with hovers.
Can you find a better name?

What about this:
```
class Debouncer extends Disposable {
	constructor(debounceTimeMs: number) {}

	runDebounced(runner: () => void) {
	}
}
```